### DISCUSSION TITLE
openRoad installation

### DISCUSSION THREAD

**Discussion by msingh9** [The-OpenROAD-Project/OpenROAD#3004] [Q&A]
I was successfully able to run cmake and then make. However, output of cmake has following two messages which seems problematic to me. How do I resolve these?

-- GUI is not enabled
-- Could NOT find VTune (missing: VTune_LIBRARIES VTune_INCLUDE_DIRS)

I ran following command
cmake .. -DCMAKE_INSTALL_PREFIX=/home/msingh/openRoad -DPYTHON_INCLUDE_DIR=$(python3 -c "import sysconfig; print(sysconfig.get_path('include'))") -DPYTHON_LIBRARY=$(python3 -c\
 "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")

I also ran make afterwards, which seems to finish. There are bunch of warnings but no error message. Where do I find the executable. I don't find it in CMAKE_INSTALL_PREFIX/bin directory.

Does it mean that installation didn't complete? 

I am looking for some help because I am stuck in the installation.

I just cleared the buffer that has log, but I can redo those steps if you need the complete log output.

**Comment by maliberty** [comment]
Vtune is unimportant unless you are a developer of OR.

The GUI suggests that you don't have Qt installed.  Did you run the dependency installer?

**Reply by msingh9** [reply]
I did run dependency installer. I didn't find any message related to Qt.

**Comment by msingh9** [comment]
Thanks Maliberty for response.

I followed the steps "https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildLocally.html" and was able to build with hiccups. 
How do I make sure Qt is installed? BTW - I am trying to install it on Linux RH8. Is it supported platform?

When I try to invoke it, I get this missing library error.
openroad -help
openroad: error while loading shared libraries: libortools.so.9: cannot open shared object file: No such file or directory

**Reply by msingh9** [reply]
I also compiled libortools.so from https://developers.google.com/optimization/install/dotnet/source_linux. But it still doesn't work. I think my installation is not working.

openroad --help
openroad: symbol lookup error: openroad: undefined symbol: _ZN4absl12lts_2022062318container_internal11kEmptyGroupE

**Comment by maliberty** [comment]
I don't know what OS you are on and you haven't included any logs.  If you look at https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/etc/DependencyInstaller.sh it you'll see it references Qt libraries.

**Comment by msingh9** [comment]
I am trying to build it on RH8. I am really stuck at this point now. I ran "sudo ./DependencyInstaller.sh" successfully. 

I can check that qt5 is installed.

[msingh@lnx-msingh cv6 /tmp]$ sudo yum -y install qt5-srpm-macros.noarch
Last metadata expiration check: 0:07:51 ago on Sun 12 Mar 2023 12:19:34 AM PST.
Package qt5-srpm-macros-5.12.5-3.el8.noarch is already installed.
Dependencies resolved.
Nothing to do.

When I run cmake to create build files, I get that "GUI is not enabled". Is there a switch I need to pass to cmake. Here is my command and it's output.

[msingh@lnx-msingh cv6 build]$ cmake .. -DPYTHON_INCLUDE_DIR=/cv6/tools2/tv2/release/Python3/3.9.0/include/python3.9 -DPYTHON_LIBRARY=/cv6/tools2/tv2/release/Python3/3.9.0/lib -Dortools_DIR=/opt/or-tools
-- OpenROAD version: v2.0-7131-g1d8d24209
-- System name: Linux
-- Compiler: GNU 8.3.1
-- Build type: RELEASE
-- Install prefix: /usr/local
-- C++ Standard: 17
-- C++ Standard Required: ON
-- C++ Extensions: OFF
-- TCL library: /usr/lib64/libtcl.so
-- TCL header: /usr/include/tcl.h
-- TCL readline library: /usr/lib64/libtclreadline.so
-- TCL readline header: /usr/include
-- Found Boost: /usr/local/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0")
-- boost: 1.80.0
-- spdlog: 1.8.1
-- STA version: 2.4.0
-- STA git sha: f84e617fc435d2d280a3f76045aa461afa386e2b
-- System name: Linux
-- Compiler: GNU 8.3.1
-- Build type: RELEASE
-- Build CXX_FLAGS: -O3 -DNDEBUG
-- Install prefix: /usr/local
-- TCL library: /usr/lib64/libtcl.so
-- TCL header: /usr/include/tcl.h
-- SSTA: 0
-- STA executable: /dump/vlsidump302/msingh/OpenROAD/src/sta/app/sta
-- GPU is not enabled
-- GUI is not enabled
-- Found Boost: /usr/local/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0") found components: serialization
-- Could NOT find VTune (missing: VTune_LIBRARIES VTune_INCLUDE_DIRS)
-- Found Boost: /usr/local/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found suitable version "1.80.0", minimum required is "1.78")
-- Found Boost: /usr/local/lib/cmake/Boost-1.80.0/BoostConfig.cmake (found version "1.80.0") found components: serialization system thread
-- TCL readline enabled
-- Tcl Extended disabled
-- Python3 disabled
-- Configuring done
-- Generating done
-- Build files have been written to: /dump/vlsidump302/msingh/OpenROAD/build

**Reply by msingh9** [reply]
I found some more information. I think QT5 installation is not correct on my machine and this is the problem. I just don't know how to get it. When I do "which qmake" I don't find it. I can find qmake-qt4.

[msingh@lnx-msingh cv6 /tmp]$ which qmake-qt4
/usr/bin/qmake-qt4

However when I run " sudo yum -y install qt5-srpm-macros.noarch" I see it says that it is already installed.

[msingh@lnx-msingh cv6 /tmp]$ sudo yum -y install qt5-srpm-macros.noarch
Last metadata expiration check: 0:54:50 ago on Sun 12 Mar 2023 12:19:34 AM PST.
Package qt5-srpm-macros-5.12.5-3.el8.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!

**Reply by vvbandeira** [reply]
You can try to remove the old version (Qt4) or pass to CMake the path to Qt5 with `-D Qt5_DIR=<PATH>`

**Comment by maliberty** [comment]
What OS are you using?  Is qt in the default location?

**Reply by msingh9** [reply]
I am using Linux Redhat Enterprise EL8. I think I am missing Qt installation.  I will try to get this fixed and will try installation. Non Gui part seems working fine so far.

**Reply by maliberty** [reply]
We don't test on RH Enterprise.  You need to make sure you have the qt development package and not just the runtime one.

**Comment by vijayank88** [comment]
@msingh9 
Can you try docker based installation [steps](https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithDocker.html) ?

**Comment by msingh9** [comment]
Thank you everyone for support. I finally managed to make it work on RH8. It is exciting to see this tool in action. I really like to appreciate everyone effort to make this open-source magic.  I will post my results once I can make it work.
